### DISCUSSION TITLE
Feature request: Import SQLite DB files or SQL-files into PocketBase

### DISCUSSION THREAD

**Discussion by christerjohansson** [pocketbase/pocketbase#1228] [Q&A]
I am struggling with demo content for PocketBase. I have the famous Northwind database in SQL, SQLite3 and a DB-file. Would be very nice to be able to import this db into PocketBase. :)

So far I have tried using DB Browser SQLite to import the DB. I can import it into the PocketBase data.db-file, but it does not show up in PocketBase.

**Comment by christerjohansson** [comment]
This is what my DB looks like currently, 
![image](https://user-images.githubusercontent.com/3787897/206797340-84604e97-7c9d-486c-888c-ae573490436e.png)

**Comment by ganigeorgiev** [comment]
While PocketBase maintain its data in regular sql tables, it needs some additional metadata in order to properly visualize it in the Admin UI. This metadata is stored in the "_collections" table.

There are also other requirements for the data tables too, like what type and values the columns are populated with. For a list with all supported PocketBase field values you can check https://pocketbase.io/docs/collections/ (_note that we use zero-defaults and `null` values are not supported, with an exception for the `json` type field_).

There are plans to allow "autodiscovery" of a table sometime in the future, but there are too many edge cases and things that could break and this will require too much time in order to make it work properly and for now is left out of the scope for v1. Partially this will be covered in the near future by the Import/Export feature in https://github.com/pocketbase/pocketbase/issues/48.

So in short - in order to have your imported tables visualized in the Admin UI, you'll have to:

1. Create "Collections" for each table (the easiest way would be from the UI).
2. Adjust the columns of the data tables to match with the [fields that PocketBase support](https://pocketbase.io/docs/collections/#overview).
